internal/service/domain/sensor/statusUpdater_test.go

```diff
@@ -0,0 +1,104 @@
+package sensor
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	storageMock "github.com/green-ecolution/green-ecolution-backend/internal/storage/_mock"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestSensorService_RunStatusUpdater(t *testing.T) {
+	t.Run("should update stale sensor statuses periodically", func(t *testing.T) {
+		ctx := context.Background()
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		svc := NewStatusUpdater(sensorRepo)
+
+		staleSensor := &entities.Sensor{
+			ID:        "sensor-1",
+			UpdatedAt: time.Now().Add(-73 * time.Hour), // 73 hours ago
+		}
+		recentSensor := &entities.Sensor{
+			ID:        "sensor-2",
+			UpdatedAt: time.Now().Add(-1 * time.Hour), // 1 hour ago
+		}
+
+		sensorRepo.EXPECT().GetAll(mock.Anything).Return([]*entities.Sensor{staleSensor, recentSensor}, nil)
+		sensorRepo.EXPECT().Update(mock.Anything, staleSensor.ID, mock.Anything).Return(staleSensor, nil)
+
+		go func() {
+			svc.RunStatusUpdater(ctx, 10*time.Millisecond)
+		}()
+
+		time.Sleep(100 * time.Millisecond)
+
+		sensorRepo.AssertCalled(t, "GetAll", mock.Anything)
+		sensorRepo.AssertCalled(t, "Update", mock.Anything, staleSensor.ID, mock.Anything)
+		sensorRepo.AssertExpectations(t) // Verifies all expectations are met
```

Comment:
Now there is it, but before there was no `AssertCalled` function called. This is fine :+1:

Reply:
Maybe I was looking at an old version of the file